### PR TITLE
Better handling of no-ether when starting without ethernet

### PR DESCRIPTION
**Pull request by masinter**
No longer wait for PUP or NS to be ready if ETHERNET-AVAILABLE returns false.
Add checks to \DEVICEEVENT -- if no network, just leave it off.
Moved CANONICAL.HOSTNAME from PUP to LLETHER because we might move to a setup where TCP is on, but PUP and XNS are not loaded.

Partly addresses #326 but doesn't fix it completely -- if you set DEFAULTPRINTINGHOST to an empty string, it calls NSPRINT which errors. The printing code is pretty complex and I'm not sure what is wanted.

I think it should be possible to have a loadup with POSTSCRIPT (and PDF?) but no INTERPRESS and PRESS.

### CODE REVIEW

**Comment by nbriggs** [comment]
It should be the case that if you don't ask for PRESS or INTERPRESS that the code doesn't get invoked.
`IL:DEFAULTPRINTERTYPE` should probably be set to `IL:POSTSCRIPT` instead of `IL:INTERPRESS` (as part of the site init file?) and the code that looks at the output stream type for hardcopy should know that an output stream type of `IL:POSTSCRIPT` suggests a file extension of `.PS` (it currently only knows INTERPRESS => IP and one other thing, perhaps PRESS=>PRESS)

File: sources/LLETHER [hunk 1]
```diff
@@ -1471,10 +1498,13 @@ Copyright (c) 1982, 1983, 1984, 1985, 1986, 1987, 1988, 1990, 1991, 1992, 1993 b
                               HOSTNAMEP _ 'NILL])
 
 (\ETHEREVENTFN
-  [LAMBDA (DEV EVENT)                                    (* ; "Edited 15-Jan-88 01:30 by bvm")
-    (SELECTQ EVENT
-        ((NIL AFTERLOGOUT AFTERSYSOUT AFTERMAKESYS AFTERSAVEVM RESTART) 
-             (PROG (NDB TURNOFFNS TIMESET)
+  [LAMBDA (DEV EVENT)                                  (* ; "Edited  1-May-2021 19:23 by larry")
+    (PROG (NDB TURNOFFNS TIMESET)
+          (OR (SUBRCALL ETHER-AVAILABLE)
```
Comment (nbriggs) [inline sources/LLETHER:1503]: I'm not convinced this is the right place to do this -- if you make a sysout on a machine with ethernet enabled, then you need to do the cleanup that follows (e.g., clearing `\PUP.READY` etc., clearing the ETHERPORTS) or you'll get the wrong behavior.

**Review by nbriggs — CHANGES_REQUESTED** [review]
see inline comment

File: sources/LLETHER (hunk 1 above)
Comment (masinter) [reply sources/LLETHER]: good catch, updated. I made a separate issue for the printer problem.

**Comment by masinter** [comment]
made suggested changes

**Pull request merged**